monai/transforms/compose.py

```diff
@@ -258,6 +261,12 @@ def randomize(self, data: Any | None = None) -> None:
                     f'Transform "{tfm_name}" in Compose not randomized\n{tfm_name}.{type_error}.', RuntimeWarning
                 )
 
+    def get_index_of_first(self, predicate):
+        for i in range(len(self.transforms)):
```

Comment:
Since we only flatten when the sub-transform is `Compose`, will the index returned here be correct, if the transform list contains `SomeOf` or `RandomOrder`?
https://github.com/Project-MONAI/MONAI/blob/0223e6262239b6390b57b630f2ceca3fea9a70bf/monai/transforms/compose.py#L279

Reply:
Seems here len only calculates the length of the transform list, so if there is `Compose` in the transform list will also return the wrong index. Does this expect?
```
from monai.transforms import Compose, NormalizeIntensity, Transform, RandomOrder

trans = Compose([
    NormalizeIntensity(),
    Flip(),
    Compose([
        NormalizeIntensity(),
        Flip()
    ]),
    RandomOrder([
        NormalizeIntensity(),
        Flip(),
    ])
])

index = trans.get_index_of_first(
            lambda t: isinstance(t, RandomizableTrait) or not isinstance(t, Transform)
        )

print(len(trans), index)  # --> 5 2
```

Reply:
Note that transforms aren't flattened on the forward pass, and it is correct that they aren't. All sorts of additional problems will start occurring if we do that.

Reply:
I'll update the documentation on the method to make it clear that it doesn't step into nested transforms when calculating indices if that helps